bench/arrow/optimized/vector_hash.cc.ll

```diff
@@ -52278,11 +52277,14 @@ for.end.i.i.i:                                    ; preds = %for.body.i.i.i
   %mul.i.i32.i.i = mul i64 %xor.i.i40.i.i.i, 1609587791953885689
   %shr.i4.i.i33.i.i = lshr i64 %mul.i.i32.i.i, 32
   %xor.i5.i.i34.i.i = xor i64 %shr.i4.i.i33.i.i, %mul.i.i32.i.i
-  %cmp203.i.i.i = icmp ugt i32 %conv.i.i.i, 143
+  %cmp203.i.i.i = icmp ugt i64 %length, 143
   br i1 %cmp203.i.i.i, label %for.body21.preheader.i.i.i, label %_ZL21XXH3_len_129to240_64bPKhmS0_mm.exit.i.i
 
 for.body21.preheader.i.i.i:                       ; preds = %for.end.i.i.i
-  %div25.i.i.i = lshr i64 %length, 4
+  %conv.i.i.i = trunc nuw i64 %length to i32
+  %div25.i.i.i = lshr i32 %conv.i.i.i, 4
+  %umax.i.i.i = tail call i32 @llvm.umax.i32(i32 %div25.i.i.i, i32 9)
+  %wide.trip.count.i.i.i = zext nneg i32 %umax.i.i.i to i64
```

Comment:
Regression.

Reply:
Duplicate of https://github.com/dtcxzyw/llvm-opt-benchmark/pull/600#discussion_r1597406008